ichnaea/util.py

```diff
@@ -32,29 +31,18 @@ def gzip_open(filename, mode, compresslevel=6):
             yield gzip_file
 
 
-def encode_gzip(data, compresslevel=6, encoding="utf-8"):
+def encode_gzip(data, compresslevel=6):
     """Encode the passed in data with gzip."""
-    if encoding and isinstance(data, str):
-        data = data.encode(encoding)
-    out = BytesIO()
-    with gzip.GzipFile(
-        None, "wb", compresslevel=compresslevel, fileobj=out
-    ) as gzip_file:
-        gzip_file.write(data)
-    return out.getvalue()
+    return gzip.compress(data, compresslevel=compresslevel)
```

Comment:
This could now be:

```
encode_gzip == gzip.compress
```

which would add access to the ``mtime`` parameter. The gzip format includes a 4-byte modification time (see [wikipedia](https://en.wikipedia.org/wiki/Gzip)), and [the ``mtime`` parameter](https://docs.python.org/3/library/gzip.html#gzip.GzipFile.mtime) allows you to pick a time rather than the current instant, so that you can get the same exact bytes in two different runs. However, this only impacts a test, which already has a workaround. 

Also, [someone in 2011](https://stackoverflow.com/a/4928651/10612) didn't like the quality of the ``gzip`` module, and thought it wasn't as good as the other standard library stuff.

I now know too much about gzip.